src/Shared/Components/ServerComponentMarker.cs

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Components
+{
+    // Represents the serialized invocation to a component.
+    // We serialize this marker into a comment in the generated
+    // HTML.
+    internal struct ServerComponentMarker
+    {
+        public const string ServerMarkerType = "server";
+
+        private ServerComponentMarker(string type, string descriptor, int? sequence, string prerenderId) : this()
+        {
+            Type = type;
+            PrerenderId = prerenderId;
+            Descriptor = descriptor;
+            Sequence = sequence;
+        }
+
+        // The order in which this component was rendered/produced
+        // on the server. It matches the number on the descriptor
+        // and is used to prevent an infinite amount of components
+        // from being rendered from the client-side.
+        public int? Sequence { get; set; }
+
+        // The marker type. Right now "server" is the only valid value.
+        public string Type { get; set; }
```

Comment:
Why do we need this?

Reply:
We want to leave room to support something like "client" in the future.